tools/test.py

```diff
@@ -1303,24 +1321,35 @@ def ProcessOptions(options):
       return False
   if options.J:
     options.j = multiprocessing.cpu_count()
+  def CheckTestMode(name, option):
+    if not option in ["run", "skip", "dontcare"]:
+      print "Unknown %s mode %s" % (name, option)
+      return False
+    return True
+  if not CheckTestMode("--flaky-tests", options.flaky_tests):
```

Comment:
just remove two lines below this and do `return CheckTestMode("--flaky-tests", options.flaky_tests)`